src/maps-api.ts

```diff
@@ -28,4 +16,28 @@ export async function getPlaceAutocomplete(key: string, address: string): Promis
             municipality,
         };
     });
-}
+};
+
+// https://developer.tomtom.com/search-api/documentation/search-service/fuzzy-search
+export const getPlaceAutocomplete = async (key: string, address: string): Promise<AutoCompleteAddressResponse[]> => {
+    if (!key) {
+        throw new Error('API key is required');
+    }
+    if (!address) {
+        throw new Error('Address is required');
+    }
+    const tomtomSearchApiEndpoint = `https://api.tomtom.com/search/2/search/${address}.json`;
+    try {
+        const autocomplete = await axios.get(tomtomSearchApiEndpoint, {
+            params: {
+                key,
+                countrySet: CountryCode.AUS,
+                limit: 100,
+            },
+        });
+
+        return mapTomTomResults(autocomplete.data.results);
+    } catch (error) {
+        throw new Error('failed to retrieve data');
+    }
+};
```

Comment:
Here, I have removed the API logic into it's own function to be called within `mapTomTomResults`. 
`getPlaceAutoComplete` now includes a check to ensure the API key is passed. I have also surrounded the axios call in a try  catch to throw a more readable error.